VirtualAssistantBusinessLogic/VirtualAssistantBusinessLogic/SparQL/XMLResponseDecoder.cs

```diff
@@ -52,30 +43,72 @@ public class XMLResponseDecoder : IResponseDecoder
                     }
                     else
                     {
-                        if (id == "_" && !results.ContainsKey(id))
-                        {
-                            results[id] = new Dictionary<string, List<string>>();
-                        }
-                        //Remove Label from key
-                        key = key.Substring(0, key.Length - "Label".Length);
-                        //If the key is new
-                        if (!results[id].ContainsKey(key))
-                        {
-                            results[id][key] = new List<string>();
-                        }
-                        //The bindings have a literal subelement
-                        xmlReader.ReadToDescendant("literal");
-                        //Read the literal value for the key
-                        string value = xmlReader.ReadElementContentAsString();
-                        //If not already added (distinct)
-                        if (!results[id][key].Contains(value))
-                        {
-                            results[id][key].Add(value);
-                        }
+                        string sanitizedKey = FindAndAddKey(results, id, key);
+                        AddValueToKey(results, xmlReader, id, sanitizedKey);
                     }
                 }
             }
             return results;
         }
+        /// <summary>
+        /// Finds the key from the XML parsing and checks whether the key already exists.
+        /// If it does not exist, it is added to results array
+        /// Either way we return the found key
```

Comment:
1 - It does not find the key, it is given as an argument.
2 - It is not added to any array, it is a dictionary
3 - We already found the key. We return a sanitised version of the key.

The method name is also a bit weird. Maybe SanitizeAndAddKey() instead